client/src/components/advancedSearch/Condition.tsx

```diff
@@ -0,0 +1,271 @@
+  },
+  {
+    value: '>',
+    label: '>',
+  },
+  {
+    value: '>=',
+    label: '>=',
+  },
+  {
+    value: '==',
+    label: '==',
+  },
+  {
+    value: '!=',
+    label: '!=',
+  },
+  {
+    value: 'in',
+    label: 'in',
+  },
+];
+
+const Condition: FC<ConditionProps> = props => {
+  const {
+    onDelete,
+    triggerChange,
+    fields = [],
+    id,
+    initData,
+    className,
+    ...others
+  } = props;
+  const [operator, setOperator] = useState(
+    initData?.op || LogicalOperators[0].value
+  );
+  const [conditionField, setConditionField] = useState<Field | any>(
+    initData?.field || fields[0] || {}
+  );
+  const [conditionValue, setConditionValue] = useState(initData?.value || '');
+  const [isValuelegal, setIsValueLegal] = useState(
+    initData?.isCorrect || false
+  );
+
+  /**
+   * Check condition's value by field's and operator's type.
+   * Trigger condition change event.
+   */
+  useEffect(() => {
+    const regInt = /^\d+$/;
+    const regFloat = /^\d+\.\d+$/;
+    const regIntInterval = /^\[\d+,\d+\]$/;
+    const regFloatInterval = /^\[\d+\.\d+,\d+\.\d+]$/;
+
+    const type = conditionField?.type;
+    const isIn = operator === 'in';
+
+    switch (type) {
+      case 'int':
+        setIsValueLegal(
+          isIn
+            ? regIntInterval.test(conditionValue)
+            : regInt.test(conditionValue)
+        );
+        break;
+      case 'float':
+        setIsValueLegal(
+          isIn
+            ? regFloatInterval.test(conditionValue)
+            : regFloat.test(conditionValue)
+        );
+        break;
+      default:
+        setIsValueLegal(false);
+        break;
+    }
+
+    triggerChange({
+      field: conditionField,
+      op: operator,
+      value: conditionValue,
+      isCorrect: isValuelegal,
+      id,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [conditionField, operator, conditionValue]);
+
+  // Trigger change event if isValuelegal changed.
+  useEffect(() => {
+    triggerChange({
+      field: conditionField,
+      op: operator,
+      value: conditionValue,
+      isCorrect: isValuelegal,
+      id,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isValuelegal]);
+
+  const classes = useStyles();
+
+  // Logic operator input change.
+  const handleOpChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOperator(event.target.value);
+  };
+  // Field Name input change.
+  const handleFieldNameChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const value = event.target.value;
+    const target = fields.find(field => field.name === value);
+    target && setConditionField(target);
+  };
+  // Value input change.
+  const handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setConditionValue(value);
+  };
+
+  return (
+    <div className={`${classes.wrapper} ${className}`} {...others}>
+      <TextField
```

Comment:
we have `CustomSelector` and `SimpleMenu` components for selecting, no need to use `Textfield` directly